### PR TITLE
packaging: `setuptools_scm` + `pyproject.toml` + `cibuildwheels`

### LINKED ISSUE #42: `pip install pysolid` support
`pysolid` is now available on PyPI (https://pypi.org/project/pysolid/). But `pip install pysolid` does not work, because a Fortran compiler is required to build the package but is not available via pip, as far as I know. It will be great if `pip install` could work!

### PR DESCRIPTION
**Pull request by jhkennedy**
This 
* reverts #41 
* updates PySolid to use [`setuptools_scm`](https://github.com/pypa/setuptools_scm) as discussed here: https://github.com/insarlab/PySolid/pull/41#issuecomment-1289707805
* converts to using `pyproject.toml` where possible as discussed here: https://github.com/insarlab/PySolid/pull/41#issuecomment-1289715721
* partly solve #42 by using [`cibuildwheels`](https://cibuildwheel.readthedocs.io/en/stable/) to push binary wheels to PyPI for macOS and Linux, but not successful for Windows yet.
* adds `dependabot` to keep GitHub Actions up to date

### CODE REVIEW

File: src/pysolid/version.py [hunk 1]
```diff
@@ -1,63 +0,0 @@
...
-release_history = (
-    Tag('0.2.4', '2022-10-24'),
-    Tag('0.2.3', '2022-10-23'),
-    Tag('0.2.2', '2022-07-20'),
-    Tag('0.2.1', '2022-01-05'),
-    Tag('0.2.0', '2021-11-10'),
-    Tag('0.1.2', '2021-02-24'),
-    Tag('0.1.1', '2021-02-01'),
-    Tag('0.1.0', '2021-01-22'),
-)
```
Comment (jhkennedy) [inline src/pysolid/version.py:23]: I removed this, but we could add it back, or something like it. I don't really know what value it provides beyond any of these:
* https://github.com/insarlab/MintPy/releases
* https://pypi.org/project/pysolid/#history
* `git log --tags --simplify-by-decoration --pretty="format:%ai %d"`

File: README.md [hunk 2]
```diff
@@ -56,10 +56,13 @@ python -m pip install -r PySolid/requirements.txt
 
 ```bash
 # option 1: use pip to install pysolid into the current environment
-# use "pip install -e" to install in the development mode
 python -m pip install PySolid
 
-# option 2: manually compile the Fortran code and setup environment variable
+# option 2: use pip to install pysolid in develop mode (editable) into the current environment
+export SETUPTOOLS_ENABLE_FEATURES="legacy-editable"
+python -m pip install -e PySolid
```
Comment (jhkennedy) [inline README.md:63]: Current support for editable/develop installs via `pyproject.toml` is beta and doesn't work for PySolid unless you export this environment variable (try it without!). It's kind of an annoyance but hopefully will go away soon. Another option is to not switch to `pyproject.toml` yet

**Comment by jhkennedy** [comment]
`clibuildwheel` is working for Linux and Mac but not windows. I tried to get it going but am not sure where to go from here and am out of time for a little while. @yunjunz feel free to try and fix it, or we could drop windows from PyPI for now (since it's available via conda) and circle back later.

File: README.md (hunk 2 above)
Comment (yunjunz) [reply README.md:63]: build from source is for advanced users, so setting an env var sounds no trouble to me at all.

File: src/pysolid/version.py (hunk 1 above)
Comment (yunjunz) [reply src/pysolid/version.py:23]: I like the idea of removing the redundant `version.py` file. While checking the version, there seems to have a discrepancy between `git describe --tags` and `setuptools_scm` as below:

```
(pysolid) yunjunz:~/tools/PySolid>$ git describe --tags
v0.2.3-14-g50d06e3
(pysolid) yunjunz:~/tools/PySolid>$ python -m setuptools_scm
0.2.4.dev14+g50d06e3
```

Any idea why?

**Comment by yunjunz** [comment]
> `cibuildwheel` is working for Linux and Mac but not windows. I tried to get it going but am not sure where to go from here and am out of time for a little while. @yunjunz feel free to try and fix it, or we could drop windows from PyPI for now (since it's available via conda) and circle back later.

Thank you @jhkennedy for the many nice changes here! Dropping "Windows" from PyPI sounds like a good compromise to me.

File: src/pysolid/version.py (hunk 1 above)
Comment (yunjunz) [reply src/pysolid/version.py:23]: Setting `version_scheme` from `guess-next-dev` to `post-release` solved the inconsistency.

**Review by yunjunz — APPROVED** [review]
Looks all good to me. Thank you @jhkennedy for the great PR!

**Comment by jhkennedy** [comment]
No problem @yunjunz ! I don't know if you want to open an issue for Windows wheels or just update #42 to reflect that's still open (it's linked and will be closed by this PR currently)

**Comment by yunjunz** [comment]
> No problem @yunjunz ! I don't know if you want to open an issue for Windows wheels or just update #42 to reflect that's still open (it's linked and will be closed by this PR currently)

Let's update #42. I have updated the description in this PR to not close #42.

**Pull request merged**